src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/ProbabilisticMap.cs

```diff
@@ -56,23 +70,123 @@ public ProbabilisticMap(ReadOnlySpan<char> values)
+            Vector256<byte> mask = Vector256.GreaterThan(index, Vector256.Create((byte)15));
+            Vector256<byte> bitMask = Vector256.ConditionalSelect(mask, bitMaskUpper, bitMaskLower);
+
+            return Vector256.Equals(bitMask & bitPositions, Vector256<byte>.Zero);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> ContainsMask16Chars(Vector128<byte> charMapLower, Vector128<byte> charMapUpper, ref char searchSpace)
+        {
+            Vector128<ushort> source0 = Vector128.LoadUnsafe(ref Unsafe.As<char, ushort>(ref searchSpace));
+            Vector128<ushort> source1 = Vector128.LoadUnsafe(ref Unsafe.As<char, ushort>(ref searchSpace), (nuint)Vector128<ushort>.Count);
+
+            Vector128<byte> sourceLower = Sse2.IsSupported
+                ? Sse2.PackUnsignedSaturate((source0 & Vector128.Create((ushort)255)).AsInt16(), (source1 & Vector128.Create((ushort)255)).AsInt16())
+                : AdvSimd.Arm64.UnzipEven(source0.AsByte(), source1.AsByte());
+
+            Vector128<byte> sourceUpper = Sse2.IsSupported
+                ? Sse2.PackUnsignedSaturate(Vector128.ShiftRightLogical(source0, 8).AsInt16(), Vector128.ShiftRightLogical(source1, 8).AsInt16())
+                : AdvSimd.Arm64.UnzipOdd(source0.AsByte(), source1.AsByte());
+
+            Vector128<byte> resultLower = IsCharBitNotSet(charMapLower, charMapUpper, sourceLower);
+            Vector128<byte> resultUpper = IsCharBitNotSet(charMapLower, charMapUpper, sourceUpper);
+
+            return ~(resultLower | resultUpper);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> IsCharBitNotSet(Vector128<byte> charMapLower, Vector128<byte> charMapUpper, Vector128<byte> values)
+        {
+            Vector128<byte> highNibble = Sse2.IsSupported
+                ? Sse2.ShiftRightLogical(values.AsInt32(), VectorizedIndexShift).AsByte() & Vector128.Create((byte)15)
+                : AdvSimd.ShiftRightLogical(values, VectorizedIndexShift);
+
+            Vector128<byte> bitPositions = Shuffle(Vector128.Create(0x8040201008040201).AsByte(), highNibble);
+
+            Vector128<byte> index = values & Vector128.Create((byte)VectorizedIndexMask);
+            Vector128<byte> bitMaskLower = Shuffle(charMapLower, index);
+            Vector128<byte> bitMaskUpper = Shuffle(charMapUpper, index - Vector128.Create((byte)16));
+            Vector128<byte> mask = Vector128.GreaterThan(index, Vector128.Create((byte)15));
+            Vector128<byte> bitMask = Vector128.ConditionalSelect(mask, bitMaskUpper, bitMaskLower);
+
+            return Vector128.Equals(bitMask & bitPositions, Vector128<byte>.Zero);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> Shuffle(Vector128<byte> vector, Vector128<byte> indices)
```

Comment:
I already see 3 helpers in BCL for this - can we unite them? e.g. internal API in Vector128

Reply:
`Vector128.NativeShuffle`? Or maybe platform?

Or would it make sense to create a different place to dump internal utils? Another example that comes to mind is the `Avx2.Permute4x64` helper that I've added for the 3rd time now.

Reply:
cc @tannergooding

Reply:
`ShuffleUnsafe` might be a better name and document it as not having deterministic behavior for out of range indices might be better (we've exclusively used `Unsafe` for non-deterministic or potentially dangerous behavior so far, `Native` is typically reserved for other things)

I'd be willing to champion it if someone opened a proposal